script/bootstrap

```diff
@@ -1,5 +1,8 @@
 #!/usr/bin/env bash
 set -e
+source script/check_go_version
```

Comment:
I'm not a fan of this because it isn't very future-proof (or past-proof, for that matter.)

If someone has a version of Go <1.0 (say, 0.7) then they will get a somewhat confusing error when running this script:
`-bash: [: : integer expression expected`

Same is true for future versions >=2.0.

I'm not committed to having it in a separate script (unless you agree that we need to handle the scenario I mentioned above), but it would be nice to keep the more permissive REGEX.